src/Tests/XPack/MachineLearning/MachineLearningIntegrationTestBase.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using Nest;
+using Tests.Framework;
+using Tests.Framework.Integration;
+using Tests.Framework.ManagedElasticsearch.Clusters;
+using Tests.Framework.MockData;
+
+namespace Tests.XPack.MachineLearning
+{
+	public abstract class MachineLearningIntegrationTestBase<TResponse, TInterface, TDescriptor, TInitializer>
+		: ApiIntegrationTestBase<XPackMachineLearningCluster, TResponse, TInterface, TDescriptor, TInitializer>
+		where TResponse : class, IResponse
+		where TDescriptor : class, TInterface
+		where TInitializer : class, TInterface
+		where TInterface : class
+	{
+		public MachineLearningIntegrationTestBase(XPackMachineLearningCluster cluster, EndpointUsage usage) : base(cluster, usage) { }
+
+		protected IPutJobResponse PutJob(IElasticClient client, string jobId)
+		{
+			var putJobResponse = client.PutJob<Metric>(jobId, f => f
+				.Description("Lab 1 - Simple example")
+				.AnalysisConfig(a => a
+					.BucketSpan(new Time("30m"))
+					.Latency("0s")
+					.Detectors(d => d.Sum(c => c.FieldName(r => r.Total)))
+				)
+				.DataDescription(d => d.TimeField(r => r.Timestamp))
+			);
+
+			if (!putJobResponse.IsValid)
+				throw new Exception("Problem setting up PutJob for integration test");
+
+			return putJobResponse;
+		}
+
+		protected IOpenJobResponse OpenJob(IElasticClient client, string jobId)
+		{
+			var openJobResponse = client.OpenJob(jobId, f => f);
+			if (!openJobResponse.IsValid || openJobResponse.Opened == false)
+				throw new Exception("Problem setting up OpenJob for integration test");
+			return openJobResponse;
+		}
+
+		protected ICloseJobResponse CloseJob(IElasticClient client, string jobId)
+		{
+			var closeJobResponse = client.CloseJob(jobId, f => f);
+			if (!closeJobResponse.IsValid || closeJobResponse.Closed == false)
+				throw new Exception("Problem setting up OpenJob for integration test");
+			return closeJobResponse;
+		}
+
+		protected IPutDatafeedResponse PutDatafeed(IElasticClient client, string jobId)
+		{
+			var putDataFeedResponse = client.PutDatafeed<Metric>(jobId + "-datafeed", f => f
+				.JobId(jobId)
+				.Indices(Nest.Indices.Parse("server-metrics"))
+				.Types(Types.Parse("metric"))
+				.Query(q => q.MatchAll(m => m.Boost(1))));
+
+			if (!putDataFeedResponse.IsValid || putDataFeedResponse.IsValid == false)
+				throw new Exception("Problem setting up PutDatafeed for integration test");
+
+			return putDataFeedResponse;
+		}
+
+		protected IStartDatafeedResponse StartDatafeed(IElasticClient client, string jobId)
+		{
+			var startDatafeedResponse = client.StartDatafeed(jobId + "-datafeed", f => f);
+			if (!startDatafeedResponse.IsValid || startDatafeedResponse.Started == false)
+				throw new Exception("Problem setting up StartDatafeed for integration test");
+			return startDatafeedResponse;
+		}
+
+		protected void IndexSnapshot(IElasticClient client, string jobId, string snapshotId)
```

Comment:
Mind commenting these, there is a lot of assumed knowledge here for the uninitiated (me!) on why we need to manually  insert these documents.